### PR TITLE
VP-44: Added false-positive Edge status check

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/VP-44

### CODE REVIEW

File: plugin/src/main/java/org/opennms/velocloud/pollers/edge/EdgeManagementStatusPoller.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.opennms.velocloud.pollers.edge;
```
Comment (aramosvizcarra) [inline plugin/src/main/java/org/opennms/velocloud/pollers/edge/EdgeManagementStatusPoller.java:1]: I think you are missing a license here

File: plugin/src/main/java/org/opennms/velocloud/pollers/edge/EdgeManagementStatusPollerFactory.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package org.opennms.velocloud.pollers.edge;
```
Comment (aramosvizcarra) [inline plugin/src/main/java/org/opennms/velocloud/pollers/edge/EdgeManagementStatusPollerFactory.java:1]: Same missing license

**Review by aramosvizcarra — CHANGES_REQUESTED** [review]
Just a couple of small changes... Otherwise Looks good to me

**Review by aramosvizcarra — APPROVED** [review]
LGTM

File: plugin/src/main/java/org/opennms/velocloud/pollers/edge/EdgeManagementStatusPollerFactory.java (hunk 2 above)
Comment (christianpape) [reply plugin/src/main/java/org/opennms/velocloud/pollers/edge/EdgeManagementStatusPollerFactory.java:29]: Fixed.

File: plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java [hunk 3]
```diff
@@ -105,6 +107,34 @@ public class VelocloudEventIngestor implements Runnable, HealthCheck {
...
+        private String enterpriseId;
+        private String alias;
+
+        public RequisitionIdentifier(final Node n) {
+            final Map<String, String> map = n.getMetaData().stream()
+                    .filter(metaData -> Objects.equals(metaData.getContext(), AbstractRequisitionProvider.VELOCLOUD_METADATA_CONTEXT))
+                    .collect(Collectors.toMap(metaData -> metaData.getKey(), metaData -> metaData.getValue()));
+            foreignSource = n.getForeignSource();
+            enterpriseId = map.get("enterpriseId");
+            alias = map.get("alias");
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java:121]: Please add non-null checks.

File: plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java [hunk 4]
```diff
@@ -136,38 +166,55 @@ public void run() {
             lastPoll = now.minus(delay, ChronoUnit.MILLIS);
         }
 
-        for (final String alias : connectionManager.getAliases()) {
+        final Set<RequisitionIdentifier> requisitionIdentifiers = nodeDao.getNodes().stream()
+                .filter(node -> node.getMetaData().stream()
+                    .anyMatch(metaData -> Objects.equals(AbstractRequisitionProvider.VELOCLOUD_METADATA_CONTEXT, metaData.getContext())))
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java:171]: Can we filter on nodes having the `alias` attribute set here?

File: plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java [hunk 5]
```diff
@@ -136,38 +166,55 @@ public void run() {
...
+                .map(node -> new RequisitionIdentifier(node))
+                .collect(Collectors.toSet());
 
-            Optional<VelocloudApiPartnerClient> velocloudApiPartnerClient = Optional.empty();
-            try {
-                velocloudApiPartnerClient = connectionManager.getPartnerClient(alias);
-            } catch (Exception e) {
-                // ignoring, maybe it is a customer connection
+        for(final RequisitionIdentifier requisitionIdentifier : requisitionIdentifiers) {
+            Optional<VelocloudApiCustomerClient> velocloudApiCustomerClient = Optional.empty();
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java:176]: Can be simplified.

File: plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java [hunk 6]
```diff
@@ -136,38 +166,55 @@ public void run() {
...
+                        LOG.debug("Cannot create customer connection for the given partner alias/enterpriseId: {}/{}", requisitionIdentifier.alias, requisitionIdentifier.enterpriseId);
+                    }
+                } catch (Exception e) {
+                    LOG.debug("Error creating customer connection for the given partner alias/enterpriseId: {}/{}", requisitionIdentifier.alias, requisitionIdentifier.enterpriseId);
+                }
+            } else {
+                try {
+                    velocloudApiCustomerClient = connectionManager.getCustomerClient(requisitionIdentifier.alias);
+                } catch (Exception e) {
+                    LOG.debug("Error creating customer connection for the given credentials: {}", requisitionIdentifier.alias);
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java:193]: Log as error?

File: plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java [hunk 7]
```diff
@@ -263,25 +310,36 @@ private void processCustomerEvents(final Instant start, final Instant end, final
...
         LOG.debug("{} enterprise events found.", customerEventList.size());
 
+        int processed = 0;
+        int ignored = 0;
         for (final CustomerEvent customerEvent : customerEventList) {
             if (!Strings.isNullOrEmpty(customerEvent.getEdgeName())) {
                 processEvent(foreignSource, customerEvent.getEdgeName(), customerEvent);
+                processed++;
             } else {
                 LOG.debug("ignoring enterprise event #{} due to missing edgeName.", customerEvent.getId());
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java:320]: Can be removed.

File: plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java [hunk 8]
```diff
@@ -263,25 +310,36 @@ private void processCustomerEvents(final Instant start, final Instant end, final
...
         LOG.debug("{} proxy events found.", enterpriseEventList.size());
+
+        int processed = 0;
+        int ignored = 0;
         for (final PartnerEvent partnerEvent : enterpriseEventList) {
             if (!Strings.isNullOrEmpty(partnerEvent.getGatewayName())) {
                 processEvent(foreignSource, partnerEvent.getGatewayName(), partnerEvent);
+                processed++;
             } else {
                 LOG.debug("ignoring proxy event #{} due to missing gatewayName.", partnerEvent.getId());
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java:338]: Same here.

File: plugin/src/main/java/org/opennms/velocloud/pollers/AbstractStatusPoller.java [hunk 9]
```diff
@@ -145,7 +146,13 @@ public VelocloudApiPartnerClient partnerClient() throws VelocloudApiException {
         }
 
         public VelocloudApiCustomerClient customerClient() throws VelocloudApiException {
-            return AbstractStatusPoller.this.clientManager.getCustomerClient(this.getClientCredentials());
+            final String enterpriseId = this.request.getPollerAttributes().get("enterpriseId");
+
+            if (Strings.isNullOrEmpty(enterpriseId)) {
+                return AbstractStatusPoller.this.clientManager.getCustomerClient(this.getClientCredentials());
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/pollers/AbstractStatusPoller.java:152]: Can we move this to the connection manager and re-use in the event processor?

**Review by fooker — CHANGES_REQUESTED** [review]

**Review by fooker — APPROVED** [review]

**Pull request merged**